web/src/pages/loading.tsx

```diff
@@ -20,7 +20,12 @@ const LoadingPage = (): ReactElement => {
   const loginPageEnabled = process.env.FEATURE_LOGIN_PAGE === "true";
 
   useEffect(() => {
-    if (!router) {
+    /**
+     * For client-side rendering of the Next.js router, the `query` property
+     * is first initialized as an empty object. This `isReady` check ensures
+     * we don't have false negatives when checking `router.query`.
+     */
+    if (!router?.isReady) {
```

Comment:
the absence of this check, combined with the catch-all `else` condition, was causing the following bug:
1. Authenticated user is redirected back from myNJ
2. [The prerendering of the page will initialize `router.query` to `{}`](https://nextjs.org/docs/pages/api-reference/functions/use-router#router-object)
3. This means the `if` condition will be false, even though they are routed to the page with a `code` query parameter
4. The `else` condition triggers sign in, which sends us back to (1) with a new code.

This situation is avoided with the `isReady` check

Reply:
[dust] You may want to take a guard clause approach here, and make the check more explicit rather than through optional chaining for readability's sake.